app/uk/gov/hmrc/tariffclassificationfrontend/service/CommodityCodeService.scala

```diff
@@ -16,26 +16,56 @@
 
 package uk.gov.hmrc.tariffclassificationfrontend.service
 
-import javax.inject.Singleton
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.CommodityCode
 
 import scala.io.Source
 
 @Singleton
-class CommodityCodeService() {
+class CommodityCodeService @Inject()(appConfig: AppConfig) {
 
+  private lazy val dateTimeFormatter = DateTimeFormatter.ofPattern("yyy-MM-dd HH:mm:ss")
   private lazy val padLimit = 10
 
-  def checkIfCodeExists(commodityCode: String): Boolean = {
-    val canonicalCode: String =
-      if (commodityCode.length > padLimit) commodityCode.substring(0, padLimit)
-      else commodityCode.trim.padTo[Char, String](padLimit, '0').mkString
+  def find(commodityCode: String): Option[CommodityCode] = {
+    val canonicalCode: String = padTo10Digits(commodityCode)
+
+    commodityCodesFromFile.find(_.code == canonicalCode)
+  }
+
+  private def padTo10Digits(input: String): String = {
+    val trimmed = input.trim
+    if (trimmed.length > padLimit) trimmed.substring(0, padLimit)
+    else trimmed.padTo[Char, String](padLimit, '0').mkString
+  }
+
+  private val commodityCodesFromFile: Seq[CommodityCode] = {
```

Comment:
Why is not lazy anymore? Do we want to read the file every time?

Reply:
It's a val so it would only be read once.
I've added lazy back on anyway